### PR TITLE
Feat/prevent russian gas imports

### PR DESCRIPTION
**Pull request by nicolezechneraggm**
closes https://github.com/AGGM-AG/pypsa-at-planning/issues/125 

## Changes proposed in this Pull Request

Contrary to earlier statements in the original PR https://github.com/AGGM-AG/pypsa-at/pull/72, Russian natural gas imports via Ukraine are not completely turned off. 
According to ENTSO-G (https://gasdashboard.entsog.eu/), Russian gas still reaches Europe via the TurkStream route. 

The current implementation features maximum modularity for stops (and restarts) of natural gas imports from Russia: 
### Two possible routes
-  `eastern_border_block` for the land-border imports via Ukrainian territories: blocks import of gas from Ukraine to neighboring countries ( default: `[FI, EE, LV, LT, PL, SK, RO]`) starting in 2025 (default). An `end_year` can also be added, but defaults to none/infinity
- `turkstream_block` for the southern import route via the Black Sea and Bulgaria: blocks imports to Bulgaria `[BG]` with default start in 2030 and a `end_year` in infinity. 

This implementation leaves the maximum possible freedom to allow or disallow imports according to current events and is therefore more future-proof than the previous implementation.  

## Checklist

**Required:**
- [x] Changes are tested locally and behave as expected.
- [x] Code and workflow changes are documented.
- [x] A brief description of the changes has been added to `Changelog.md`.
- [x] All Sourcery Bot review suggestions have been implemented or discarded with an explanation.
- [ ] All github actions succeed.

**If applicable:**
- [ ] Changes in configuration options are reflected in `scripts/lib/validation`.
- [ ] For new data sources or versions, [these instructions](https://pypsa-eur.readthedocs.io/en/latest/data_sources.html) have been followed.
- [ ] New rules are documented in the appropriate `docs-at/` files.

## Summary by Sourcery

Introduce configurable blocking of Russian gas imports via Ukrainian land borders and TurkStream and integrate it into gas network preprocessing.

New Features:
- Add a modular block_russian_gas_imports mechanism to disable gas pipeline import capacities by country and time window based on configuration.

Enhancements:
- Replace the previous hardcoded Ukrainian transit stop logic with a generalized route-based blockade of Russian gas imports and update gas network modification workflow accordingly.

Documentation:
- Update Austrian changelog entry to describe blocking of Russian methane imports via Ukraine and TurkStream and remove outdated Ukrainian transit documentation.

Tests:
- Add tests verifying that configured Russian gas import blockades zero and freeze relevant gas pipeline import generators across planning horizons.

Chores:
- Remove obsolete Ukrainian gas transit stop data, config wiring, and exploratory gas network investigation script, and adjust the default run prefix for the new scenario.

### CODE REVIEW

File: mods/network/gas.py [hunk 1]
```diff
@@ -99,83 +99,75 @@ def unravel_gas_import_and_production(
...
+        for cc in active_countries:
+            generator_name = f"{cc} gas pipeline import"
+            if generator_name not in n.generators.index:
+                logger.warning(f"Generator '{generator_name}' not found, skipping.")
+                continue
+            n.generators.loc[generator_name, "p_nom"] = 0
+            n.generators.loc[generator_name, "p_nom_extendable"] = False
+            logger.info(
+                f"Blocked Russian gas import at '{generator_name}' ({block_name})."
+            )
```
[the remark this answers is not in the document]
Comment (nicolezechneraggm) [reply mods/network/gas.py:184]: was added with a less strict logic manually
